ksql-engine/src/test/java/io/confluent/ksql/function/udaf/topkdistinct/IntTopkDistinctKudafTest.java

```diff
@@ -134,4 +134,48 @@ public void shouldBeThreadSafe() {
     // Then:
     assertThat(result, is(new Object[]{83, 82, 81, 80, 73, 72, 71, 70, 63, 62, 61, 60}));
   }
+
+  @SuppressWarnings("unchecked")
+  //@Test
+  public void testAggregatePerformance() {
+    final int iterations = 1_000_000_000;
+    final int topX = 10;
+    final TopkDistinctKudaf<Integer> intTopkDistinctKudaf =
+        new TopkDistinctKudaf<>(0, topX, Integer.class);
+    final Integer[] aggregate = IntStream.range(0, topX)
+        .mapToObj(idx -> null)
+        .toArray(Integer[]::new);
+    final long start = System.currentTimeMillis();
+
+    for(int i = 0; i != iterations; ++i) {
+      intTopkDistinctKudaf.aggregate(i, aggregate);
```

Comment:
Does this test take advantage of your optimization? The check for whether we can skip the sort will always be false right?

Reply:
Nope, I was testing worst case to make sure I was improving things!